pkg/controller/registrysyncer/registrysyncer.go

```diff
@@ -247,6 +259,43 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, log *
 	return nil
 }
 
+func finalizeIfNeeded(ctx context.Context, stream *imagev1.ImageStream, srcClusterName string, clients map[string]ctrlruntimeclient.Client) error {
+	finalizerSet := sets.NewString(stream.ObjectMeta.Finalizers...)
```

Comment:
Nit: There is no need to do the extra hoop via ObjectMeta, just write `stream.Finalizers` (multiple places)